cinema/serializers.py

```diff
@@ -68,13 +80,67 @@ class Meta:
             "movie_title",
             "cinema_hall_name",
             "cinema_hall_capacity",
+            "tickets_available"
         )
 
 
+class TicketSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Ticket
+        fields = ("id", "row", "seat", "movie_session")
+
+    def validate(self, attrs):
+        Ticket.validate_seat_and_row(
+            attrs["seat"],
+            attrs["movie_session"].cinema_hall.seats_in_row,
+            attrs["row"],
+            attrs["movie_session"].cinema_hall.rows,
+            serializers.ValidationError,
+        )
+        return attrs
+
+
 class MovieSessionDetailSerializer(MovieSessionSerializer):
     movie = MovieListSerializer(many=False, read_only=True)
     cinema_hall = CinemaHallSerializer(many=False, read_only=True)
+    taken_places = SerializerMethodField()
 
     class Meta:
         model = MovieSession
-        fields = ("id", "show_time", "movie", "cinema_hall")
+        fields = ("id", "show_time", "movie", "cinema_hall", "taken_places")
+
+    def get_taken_places(self, movie_session):
+        return [
+            {"row": ticket.row, "seat": ticket.seat}
+            for ticket in movie_session.tickets.all()
+        ]
+
+
+class TicketRetrieveSerializer(TicketSerializer):
+    movie_session = MovieSessionListSerializer(many=False, read_only=True)
+
+
+class OrderSetPagination(PageNumberPagination):
+    page_size = 1
+    page_size_query_param = "page_size"
+    max_page_size = 10
```

Comment:
I think it will be much better to replace this class into a separate module (paginators.py in the same directory ofc)